static/app/components/modals/widgetViewerModal/widgetViewerTableCell.tsx

```diff
@@ -159,6 +165,56 @@ export const renderGridBodyCell =
     );
   };
 
+export const renderPrependColumns =
+  ({location, organization, tableData, eventView}: Props & {eventView: EventView}) =>
+  (isHeader: boolean, dataRow?: any, rowIndex?: number): React.ReactNode[] => {
+    if (isHeader) {
+      return [
+        <PrependHeader key="header-event-id">
+          <SortLink
+            align="left"
+            title={t('event id')}
+            direction={undefined}
+            canSort={false}
+            generateSortLink={() => undefined}
+          />
+        </PrependHeader>,
+      ];
+    }
+    let value = dataRow.id;
+
+    if (tableData?.meta) {
+      const fieldRenderer = getFieldRenderer('id', tableData?.meta);
+      value = fieldRenderer(dataRow, {organization, location});
+    }
+
+    const eventSlug = generateEventSlug(dataRow);
+
+    const target = eventDetailsRouteWithEventView({
+      orgSlug: organization.slug,
+      eventSlug,
+      eventView,
+    });
+
+    return [
+      <Tooltip key={`eventlink${rowIndex}`} title={t('View Event')}>
+        <StyledLink data-test-id="view-event" to={target}>
+          {value}
+        </StyledLink>
+      </Tooltip>,
+    ];
+  };
+
 const StyledTooltip = styled(Tooltip)`
   display: initial;
 `;
+
+const PrependHeader = styled('span')`
+  color: ${p => p.theme.subText};
+`;
+
+const StyledLink = styled(Link)`
```

Comment:
What happens if we just use `Link`? I removed this and used `Link` and it looks fine for me

Reply:
good question Nar! can value be also an element of just a text/number/boolean? Unfortunately atm it is of type any